fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/LoadCommand.java

```diff
@@ -0,0 +1,438 @@
+        this.sourceInfos = Objects.requireNonNull(sourceInfos, "sourceInfos should not null");
+        this.properties = Objects.requireNonNull(properties, "properties should not null");
+        this.bulkStorageDesc = Objects.requireNonNull(bulkStorageDesc, "bulkStorageDesc should not null");
+        this.comment = Objects.requireNonNull(comment, "comment should not null");
+    }
+
+    /**
+     * for test print
+     * @param ctx context
+     * @return parsed insert into plan
+     */
+    @VisibleForTesting
+    public List<LogicalPlan> parseToInsertIntoPlan(ConnectContext ctx) throws AnalysisException {
+        ctx.getState().setNereids(true);
+        List<LogicalPlan> plans = new ArrayList<>();
+        for (BulkLoadDataDesc dataDesc : sourceInfos) {
+            plans.add(completeQueryPlan(ctx, dataDesc));
+        }
+        return plans;
+    }
+
+    @Override
+    public void run(ConnectContext ctx, StmtExecutor executor) throws Exception {
+        ctx.getState().setNereids(true);
+        this.profile = new Profile("Query", ctx.getSessionVariable().enableProfile);
+        // TODO: begin txn form multi insert sql
+        profile.getSummaryProfile().setQueryBeginTime();
+        for (BulkLoadDataDesc dataDesc : sourceInfos) {
+            plans.add(new InsertIntoTableCommand(completeQueryPlan(ctx, dataDesc), Optional.of(labelName), false));
+        }
+        profile.getSummaryProfile().setQueryPlanFinishTime();
+        executeInsertStmtPlan(ctx, executor, plans);
+    }
+
+    private LogicalPlan completeQueryPlan(ConnectContext ctx, BulkLoadDataDesc dataDesc)
+            throws AnalysisException {
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("nereids load stmt before conversion: {}", dataDesc.toSql());
+        }
+        // build source columns, add select and insert node.
+        List<String> sinkCols = new ArrayList<>();
+        // contains the map of olap column to FileFieldNames, ColumnMappingList, ColumnsFromPath.
+        Map<String, NamedExpression> originSinkToSourceMap = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
+        OlapTable olapTable = getOlapTable(ctx, dataDesc);
+        Map<String, String> tvfProperties = getTvfProperties(dataDesc, bulkStorageDesc);
+
+        buildSinkColumns(olapTable, dataDesc, tvfProperties, sinkCols, originSinkToSourceMap);
+        // mapping sink colum to mapping expression
+        if (!dataDesc.getColumnMappingList().isEmpty()) {
+            for (Expression mappingExpr : dataDesc.getColumnMappingList()) {
+                if (!(mappingExpr instanceof EqualTo)) {
+                    continue;
+                }
+                EqualTo equalTo = (EqualTo) mappingExpr;
+                if (!(equalTo.left() instanceof UnboundSlot)) {
+                    // if not column, skip
+                    continue;
+                }
+                String mappingColumn = equalTo.left().toSql();
```

Comment:
why use toSql as key? we should use Expression as key directly. if we could not use expression as key. we should use normalized name to ensure equals work fine